### PR TITLE
remove recursive call from open_or_close()

### PR DESCRIPTION
**Pull request by MarcusAseth**
#### Summary
SUMMARY: Performance "remove recursive call from open_or_close() ."

#### Purpose of change
`open_or_close()` was implemented in terms of a recursive call function to itself

#### Describe the solution
`open_or_close()` is now implemented by using `find_lines_of_parts()`

#### Additional context
Also added a function comment to `open_or_close()`
Also corrected a typo in the function comment of `close()`

### CODE REVIEW

File: src/vehicle_use.cpp [hunk 1]
```diff
@@ -1191,7 +1191,7 @@ void vehicle::open( int part_index )
 }
 
 /**
- * Opens an openable part at the specified index. If it's a multipart, opens
+ * Closes an openable part at the specified index. If it's a multipart, opens
```
Comment (Night-Pryanik) [inline src/vehicle_use.cpp]:
```suggestion
 * Closes an openable part at the specified index. If it's a multipart, closes
```

File: src/vehicle_use.cpp [hunk 2]
```diff
@@ -1223,40 +1223,27 @@ void vehicle::open_all_at( int p )
...
-        //Look for parts 1 square off in any cardinal direction
-        const int dx = vp.mount().x - origin.x;
-        const int dy = vp.mount().y - origin.y;
-        const int delta = dx * dx + dy * dy;
-
-        const bool is_near = ( delta == 1 );
-        const bool is_id = part_info( next_index ).get_id() == part_info( part_index ).get_id();
-        const bool do_next = !!vp.part().open ^ opening;
+    //find_lines_of_parts() doesn't return the part_index we passed, so we set it on it's own
+    parts[part_index].open = opening;
```
Comment (mlangsdorf) [inline src/vehicle_use.cpp]: Please don't randomly churn code.  Move this back to where it started.

Comment (MarcusAseth) [reply src/vehicle_use.cpp]: done ( I hope, still figuring this out )

**Comment by Epictyphlosion** [comment]
Now we just need to fix the constant "on_load()" thing that freezes the game up after every 8 or so turns

**Pull request merged**